x-pack/plugins/observability/public/pages/alerts/parse_alert.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  RULE_ID,
+  RULE_NAME,
+  ALERT_STATUS,
+  ALERT_START,
+} from '@kbn/rule-data-utils/target/technical_field_names';
+import type { TopAlert } from '.';
+import { parseTechnicalFields } from '../../../../rule_registry/common/parse_technical_fields';
+import { asDuration, asPercent } from '../../../common/utils/formatters';
+import { ObservabilityRuleTypeRegistry } from '../../rules/create_observability_rule_type_registry';
+
+export const parseAlert = (observabilityRuleTypeRegistry: ObservabilityRuleTypeRegistry) => (
+  alert: Record<string, unknown>
+): TopAlert => {
+  const parsedFields = parseTechnicalFields(alert);
+  const formatter = observabilityRuleTypeRegistry.getFormatter(parsedFields[RULE_ID]!);
+  const formatted = {
+    link: undefined,
+    reason: parsedFields[RULE_NAME]!,
+    ...(formatter?.({ fields: parsedFields, formatters: { asDuration, asPercent } }) ?? {}),
+  };
+
+  return {
+    ...formatted,
+    fields: parsedFields,
+    active: parsedFields[ALERT_STATUS] !== 'closed',
+    start: new Date(parsedFields[ALERT_START]!).getTime(),
```

Comment:
I assume we ensure somewhere else that `parsedFields[ALERT_START]` always exists

Reply:
Good question. This is just a renamed variant of the `decorateResponse`, but I'll double-check.